src/Parser/Warrior/Arms/Modules/Core/ColossusSmash.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { formatPercentage } from 'common/format';
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import Wrapper from 'common/Wrapper';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SpellUsable from '../Features/SpellUsable';
+
+class ColossusSmashAnalyzer extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  colossusSmashes = 0;
+  shatteredColossusOverlaps = 0;
+
+  on_byPlayer_cast(event) {
+    if(SPELLS.COLOSSUS_SMASH.id === event.ability.guid) {
+      this.colossusSmashes += 1;
+      // Ignore the cast if the player didn't have shattered defenses.
+      if(!this.combatants.selected.hasBuff(SPELLS.SHATTERED_DEFENSES.id, event.timestamp - 1)) {
```

Comment:
Just out of curiosity, why `-1` on the timestamp?

Reply:
If shattered defenses is applied using warbringer instead of colossus smash the buff occurs in the same frame, so I used a 1ms offset to avoid that.

Reply:
Oh gotcha, forgot that warbreaker applies the CS debuff